packages/@lwc/engine/src/framework/vm.ts

```diff
@@ -247,6 +257,12 @@ export function createVM(elm: HTMLElement, Ctor: ComponentConstructor, options:
     // link component to the wire service
     const initializedVm = uninitializedVm as VM;
     linkComponent(initializedVm);
+
+    if (process.env.NODE_ENV !== 'production') {
```

Comment:
From a readability perspective, adding restrictions on the element in the `createVM` is not intuitive. 
How about adding these restrictions as part of the BaseLightingElementConstructor? https://github.com/salesforce/lwc/blob/master/packages/@lwc/engine/src/framework/base-lightning-element.ts#L290-L293

Reply:
I will probably agree with this. So the restrictions can kick in even if you attempt to use the methods in your constructor after the `super()` call

Reply:
Good observation @ravijayaramappa!